manifests/known_hosts.pp

```diff
@@ -25,35 +25,38 @@
 #
 # @param ssh_known_hosts_file
 #   Location of the ssh_known_hosts file (usually /etc/ssh/ssh_known_hosts).
+#   Set this to undef/null if no ssh_known_hosts file should be managed.
 #
 class profile_hostbased_ssh::known_hosts (
-  Hash   $hosts_data,
-  String $ssh_known_hosts_file,
+  Hash             $hosts_data,
+  Optional[String] $ssh_known_hosts_file,
 ) {
-  # ensure proper perms on known hosts file
-  file { $ssh_known_hosts_file :
-    ensure => file,
-    group  => 'root',
-    mode   => '0644',
-    owner  => 'root',
-  }
+  if $ssh_known_hosts_file {
+    # ensure proper perms on known hosts file
+    file { $ssh_known_hosts_file :
+      ensure => file,
+      group  => 'root',
+      mode   => '0644',
+      owner  => 'root',
+    }
 
-  # create sets of entries in /etc/ssh/ssh_known_hosts from each host set
-  $hosts_data.each | $host_set, $data | {
-    $domain   = $data[domain]
-    $key_type = $data[key_type]
-    $key      = $data[key]
-    $hosts    = $data[hosts]
-    # create individual sshkey resources from each host
-    $hosts.each | $host, $ip| {
-      $fqdn    = "${host}.${domain}"
-      $aliases = [$fqdn, $ip]
-      sshkey { $host :
-        ensure       => present,
-        host_aliases => $aliases,
-        key          => $key,
-        target       => $ssh_known_hosts_file,
-        type         => $key_type,
+    # create sets of entries in /etc/ssh/ssh_known_hosts from each host set
+    $hosts_data.each | $host_set, $data | {
```

Comment:
I do not want to slow this down, but I'm kind of surprised you made `$ssh_known_hosts_file` be the optional parameter and conditional. My gut would have been to have made the `$hosts_data` be optional and only update the `$ssh_known_hosts_file` if `$hosts_data` was not empty/undefined.

I guess I don't see much difference between the two, except how I would think about using the module. My gut is that I'd tend to think more about not supplying any `$hosts_data` by default.

Reply:
Good point, it was a weird choice on my part. I've adjusted per your suggestion.